### PR TITLE
Added CentCom pAI

### PR DESCRIPTION
**Pull request by Brahex**
## About the PR

Added a CentCom pAI as requested in [Issue75](https://github.com/echo-station/echo-station/issues/75).

## Why / Balance

Because it's funny and was requested.

As for balance, a CentCom pAI behaves almost the same as a normal pAI except for its ability to electrocute everyone near them if they so desire (once every 15 seconds). They can also give themselves All Access for 30 seconds, making it so that whoever is holding them can go wherever they want in the station! The also have access to almost all radio channels.

## Technical details

Looks different, can electrocute, can turn on All Access, has access to most radio channels, has a different name prefix when name changed.

## Media

tmp
tmp2
tmp3

## Requirements

- [X] I have read and I am following the [Pull Request Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html). I understand that not doing so may get my pr closed at maintainer’s discretion
- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

None that I'm aware of.

**Changelog**

:cl:
- add: Added a CentCom pAI.

### CODE REVIEW

File: Resources/Prototypes/_Echo/Entities/Objects/Fun/centcom_pai.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  components:
+  - type: InstantAction
+    checkCanInteract: false
+    checkConsciousness: false
+    itemIconStyle: BigAction
+    priority: 20
+    icon:
+      sprite: _Echo/Interface/Actions/pai-shock.rsi
+      state: pai-shock
+    event: !type:PAIElectrocuteActionEvent {}
```
Comment (redmushie) [inline Resources/Prototypes/_Echo/Entities/Objects/Fun/centcom_pai.yml]: There should be a reasonable cooldown. `InstantAction` has a cooldown setting.

File: Content.Server/_Echo/PAI/PAICentComSystem.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    /// </summary>
+    /// <param name="uid"></param>
+    /// <param name="component"></param>
+    /// <param name="args"></param>
+    private void OnElectrocuteAction(EntityUid uid, PAICentComComponent component, PAIElectrocuteActionEvent args)
+    {
+        foreach (var targetUid in _lookup.GetEntitiesInRange(_transform.GetMapCoordinates(uid), 0.2f))
+            _electrocution.TryDoElectrocution(targetUid, uid, 5, TimeSpan.FromSeconds(3), false);
+        _audio.PlayPvs(component.Sound, uid);
+        _actionsSystem.SetCooldown(args.Action, TimeSpan.FromSeconds(20));
```
Comment (redmushie) [inline Content.Server/_Echo/PAI/PAICentComSystem.cs]: Remove this line and use the InstantAction cooldown setting instead

**Review by redmushie — CHANGES_REQUESTED** [review]
It does still assign the standard pAI naming. Is that desirable?

image

**Comment by redmushie** [comment]
Also, as is visible in the image, it should have a built-in CentCom and probably Command radio key.

File: Resources/Prototypes/_Echo/Entities/Objects/Fun/centcom_pai.yml (hunk 1 above)
Comment (Brahex) [reply Resources/Prototypes/_Echo/Entities/Objects/Fun/centcom_pai.yml]: InstantAction's cooldown is not a datafield nor is it used anywhere else in the solution. How exactly do you want me to do this?

Comment (redmushie) [reply Resources/Prototypes/_Echo/Entities/Objects/Fun/centcom_pai.yml]: It's called `useDelay`.

**Comment by Brahex** [comment]
TIL you can put a pAI inside a PDA

File: Resources/Prototypes/_Echo/Entities/Objects/Fun/centcom_pai.yml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  - type: ToggleableGhostRole
+    roleName: pai-system-role-name-centcom
+    roleDescription: pai-system-role-description-centcom
+    roleRules: ghost-role-information-familiar-rules
+  - type: Access
+    groups:
+    - AllAccess
+  - type: Tag
+    tags:
+    - DoorBumpOpener
```
Comment (redmushie) [inline Resources/Prototypes/_Echo/Entities/Objects/Fun/centcom_pai.yml]: I think you misunderstood, when we said "master key" we meant the "station master encryption key".

TL;DR, Remove this, and set the radio channels to:
  - Binary
  - CentCom
  - Common
  - Command
  - Engineering
  - Medical
  - Science
  - Security
  - Service
  - Supply
  - Prison

File: Resources/Prototypes/_Echo/Entities/Objects/Fun/centcom_pai.yml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  - type: InstantAction
+    checkCanInteract: false
+    checkConsciousness: false
+    itemIconStyle: BigAction
+    priority: 20
+    icon:
+      sprite: _Echo/Interface/Actions/pai-shock.rsi
+      state: pai-shock
+    event: !type:PAIElectrocuteActionEvent
+    useDelay: 60
```
Comment (redmushie) [inline Resources/Prototypes/_Echo/Entities/Objects/Fun/centcom_pai.yml]: I think 15 seconds should be okay. If someone really keeps picking up the pAI to try to kill it it should be able to defend itself reasonably well with that.

**Review by redmushie — CHANGES_REQUESTED** [review]

File: Content.Server/PAI/PAISystem.cs [hunk 5]
```diff
@@ -56,7 +56,7 @@ private void OnRenameAction(EntityUid uid, PAIComponent component, PAIRenameActi
 
         _quickDialog.OpenDialog(actor.PlayerSession, "Rename", "Name", (string newName) =>
         {
-            _metaSystem.SetEntityName(args.Performer, $"pAI {newName}");
+            _metaSystem.SetEntityName(args.Performer, $"{Loc.GetString(component.NamePrefix)} {newName}");
```
Comment (redmushie) [inline Content.Server/PAI/PAISystem.cs]: Add Echo marker, like: `// Echo`

File: Content.Server/PAI/PAISystem.cs [hunk 6]
```diff
@@ -73,7 +73,7 @@ private void OnMindAdded(EntityUid uid, PAIComponent component, MindAddedMessage
             return;
 
         // Ownership tag
-        var val = Loc.GetString("pai-system-pai-name", ("owner", component.LastUser));
+        var val = Loc.GetString(Loc.GetString(component.PAIName), ("owner", component.LastUser));
```
Comment (redmushie) [inline Content.Server/PAI/PAISystem.cs]: Marker

File: Content.Server/PAI/PAISystem.cs [hunk 7]
```diff
@@ -120,7 +120,7 @@ private void ScrambleName(EntityUid uid, PAIComponent comp)
         }
 
         // add 's pAI to the scrambled name
-        var val = Loc.GetString("pai-system-pai-name-raw", ("name", name.ToString()));
+        var val = Loc.GetString(Loc.GetString(comp.PAINameRaw), ("name", name.ToString()));
```
Comment (redmushie) [inline Content.Server/PAI/PAISystem.cs]: Marker

File: Content.Server/_Echo/PAI/PAICentComSystem.cs [hunk 8]
```diff
@@ -0,0 +1,41 @@
+using Content.Shared._Echo.PAI;
+using Content.Server.Actions;
+using Robust.Shared.Audio.Systems;
+using Content.Server.Electrocution;
+using Robust.Server.GameObjects;
+
+namespace Content.Server._Echo.PAI;
+
+public sealed class PAICentComSystem : SharedPAICentComSystem
```
Comment (redmushie) [inline Content.Server/_Echo/PAI/PAICentComSystem.cs:11]: Add a summary, start it with "Echo: ".

**Review by redmushie — CHANGES_REQUESTED** [review]

**Comment by redmushie** [comment]
After those I'm happy to merge!

File: Content.Server/_Echo/PAI/PAICentComSystem.cs (hunk 8 above)
Comment (Brahex) [reply Content.Server/_Echo/PAI/PAICentComSystem.cs:11]: Just double checking: You sure you want the marker in files that are already in _Echo folders?

File: Resources/Prototypes/_Echo/Entities/Objects/Fun/centcom_pai.yml (hunk 3 above)
Comment (Brahex) [reply Resources/Prototypes/_Echo/Entities/Objects/Fun/centcom_pai.yml]: Was already working on making all access an action to prevent abuse and wanton killing of little innocent pAI's. Also, it'll make the pAI player feel usefull.

Comment (Brahex) [reply Resources/Prototypes/_Echo/Entities/Objects/Fun/centcom_pai.yml]: Give them Binary and CentCom as intrinsic. All others are now encryption keys.

File: Content.Server/_Echo/PAI/PAICentComSystem.cs (hunk 8 above)
Comment (redmushie) [reply Content.Server/_Echo/PAI/PAICentComSystem.cs:11]: Hadn't noticed it was in a seperate folder. Should be fine!

**Review by redmushie — APPROVED** [review]
